File: generator/src/main/java/ch/bailu/gtk/table/NamespaceTable.java
package ch.bailu.gtk.table;

import java.util.HashMap;
import java.util.Map;

public class NamespaceTable {
    private final Map<String, String> table = new HashMap<>();


    private final static NamespaceTable INSTANCE = new NamespaceTable();

    public static NamespaceTable instance() {
        return INSTANCE;
    }

    private NamespaceTable() {}

    public void add(String namespace) {
        table.put(namespace.toLowerCase(), namespace);
    }


    public boolean contains(String namespace) {
        return table.containsKey(namespace);
    } 
}

File: generator/src/main/kotlin/ch/bailu/gtk/table/NamespaceTable.kt
package ch.bailu.gtk.table

import ch.bailu.gtk.config.NamespaceConfig
import java.util.*

object NamespaceTable {
    private val table: MutableMap<String, NamespaceConfig> = HashMap()

    fun add(namespace: String, namespaceConfig: NamespaceConfig) {
        table[namespace.lowercase()] = namespaceConfig
    }

    operator fun contains(namespace: String): Boolean {
        return table.containsKey(namespace)
    }

    fun with(namespace: String, action: (NamespaceConfig) -> Unit) {
        val config = table.get(namespace)
        if (config != null) {
            action(config)
        }
    }
}